instrumentation/opentelemetry-instrumentation-flask/tests/test_programmatic.py

```diff
@@ -459,6 +459,7 @@ def test_exclude_lists_from_explicit(self):
         self.assertEqual(len(span_list), 1)
 
     def test_flask_metrics(self):
+        _server_duration_attrs_old.append("http.route")
```

Comment:
IMO `_server_duration_attrs_old` itself should be updated at the source. If that's not an option, I suggest that `_recommended_metrics_attrs_old` and `_recommended_metrics_attrs_both` use some copy of `_server_duration_attrs_old` that includes `http.route`. Modifying this global state here could easily break other tests in confusing ways.

Reply:
I agree with you on changing the _server_duration_attrs_old at the source, but they are imported from the semconv package, and I cannot modify them. I committed the changes and made a copy for the test. Please let me know if this is what you intended

Reply:
> I agree with you on changing the _server_duration_attrs_old at the source, but they are imported from the semconv package, and I cannot modify them.

It's in this same repo, so I *think* you could change them in this PR. I've done something similar in https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2624. But maybe that won't be accepted.

Your `_server_duration_attrs_old_copy` is an OK solution for the sake of this test, but the line being commented on here (`_server_duration_attrs_old.append("http.route")`) still needs to be removed.

Reply:
Sorry, I may have misunderstood, but if I remove the line `_server_duration_attrs_old.append("http.route")`, the test where I assert to have the `http.route` as a metric attribute will fail (`test_basic_metric_success`, for example). I can also remove the `http.route` from the assertion list, and this will result in a successful run for all tests. Is that the solution you're looking for?

Reply:
`test_basic_metric_success` should pass if you run it by itself, it shouldn't depend on another test. Sounds like there's a bug in the implementation.

Reply:
My mistake!! i was having troubles with the `tox` in local. Now its done :)